src/LiveComponent/tests/Integration/LiveComponentHydratorTest.php

```diff
@@ -144,6 +144,140 @@ public function testCanDehydrateAndHydrateComponentWithTestCases(callable $testF
 
     public function provideDehydrationHydrationTests(): iterable
     {
+        yield 'onUpdated: exception if method not exists' => [function () {
+            return HydrationTest::create(new class() {
+                #[LiveProp(writable: true, onUpdated: 'onFirstNameUpdated')]
+                public string $firstName;
+            })
+                ->mountWith(['firstName' => 'Ryan'])
+                ->userUpdatesProps(['firstName' => 'Victor'])
+                ->expectsExceptionDuringHydration(\Exception::class, '/onFirstNameUpdated\(\)" specified as LiveProp "onUpdated" hook does not exist/');
+        }];
+
+        yield 'onUpdated: with string value' => [function () {
+            return HydrationTest::create(new class() {
+                #[LiveProp(writable: true, onUpdated: 'onFirstNameUpdated')]
+                public string $firstName;
+
+                public function onFirstNameUpdated($oldValue)
+                {
+                    if ('Victor' === $this->firstName) {
+                        $this->firstName = 'Revert to '.$oldValue;
+                    }
+                }
+            })
+                ->mountWith(['firstName' => 'Ryan'])
+                ->userUpdatesProps(['firstName' => 'Victor'])
+                ->assertObjectAfterHydration(function (object $object) {
+                    $this->assertSame('Revert to Ryan', $object->firstName);
+                });
+        }];
+
+        yield 'onUpdated: with integer value' => [function () {
+            return HydrationTest::create(new class() {
+                #[LiveProp(writable: true, onUpdated: 'onNumberUpdated')]
+                public int $number;
+
+                public function onNumberUpdated($oldValue)
+                {
+                    if (9 === $this->number) {
+                        $this->number = $oldValue;
+                    }
+                }
+            })
+                ->mountWith(['number' => 10])
+                ->userUpdatesProps(['number' => 9])
+                ->assertObjectAfterHydration(function (object $object) {
+                    $this->assertSame(10, $object->number);
+                });
+        }];
+
+        yield 'onUpdated: with object value' => [function () {
```

Comment:
```suggestion
        yield 'onUpdated: set to an array' => [function () {
```

(really, the important part is that you're able to add a hook for specific writable paths - technically the property value could be an array)